cpp/benchmarks/string/url_decode.cu

```diff
@@ -16,55 +16,68 @@
 
 #include <benchmark/benchmark.h>
 #include <benchmarks/fixture/benchmark_fixture.hpp>
-#include <benchmarks/fixture/templated_benchmark_fixture.hpp>
 #include <benchmarks/synchronization/synchronization.hpp>
 
 #include <cudf/column/column_view.hpp>
+#include <cudf/filling.hpp>
 #include <cudf/strings/convert/convert_urls.hpp>
+#include <cudf/strings/strings_column_view.hpp>
 #include <cudf/types.hpp>
 
 #include <cudf_test/base_fixture.hpp>
 #include <cudf_test/column_utilities.hpp>
 #include <cudf_test/column_wrapper.hpp>
 #include <cudf_test/cudf_gtest.hpp>
 
-#include <algorithm>
-#include <random>
+#include <thrust/execution_policy.h>
+#include <thrust/for_each.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/random.h>
 
 struct url_string_generator {
-  size_t num_chars;
-  std::bernoulli_distribution dist;
-
-  url_string_generator(size_t num_chars, double esc_seq_chance)
-    : num_chars{num_chars}, dist{esc_seq_chance}
+  char* chars;
+  double esc_seq_chance;
+  thrust::minstd_rand engine;
+  thrust::uniform_real_distribution<float> esc_seq_dist;
+  url_string_generator(char* c, double esc_seq_chance, thrust::minstd_rand& engine)
+    : chars(c), esc_seq_chance(esc_seq_chance), engine(engine), esc_seq_dist(0, 1)
   {
   }
 
-  std::string operator()(std::mt19937& engine)
+  __device__ void operator()(thrust::tuple<cudf::size_type, cudf::size_type> str_begin_end)
   {
-    std::string str;
-    str.reserve(num_chars);
-    while (str.size() < num_chars) {
-      if (str.size() < num_chars - 3 && dist(engine)) {
-        str += "%20";
+    auto begin = thrust::get<0>(str_begin_end);
+    auto end   = thrust::get<1>(str_begin_end);
+    engine.discard(begin);
+    for (auto i = begin; i < end; ++i) {
+      if (esc_seq_dist(engine) < esc_seq_chance and i < end - 3) {
+        chars[i]     = '%';
+        chars[i + 1] = '2';
+        chars[i + 2] = '0';
+        i += 2;
       } else {
-        str.push_back('a');
+        chars[i] = 'a';  // + (i % 26);
```

Comment:
```suggestion
        chars[i] = 'a';
```

Reply:
Kept it here as a hint for suggestion during review. will remove this.

Reply:
Are you suggesting that the line should be `chars[i] = 'a' + (i % 26)` or is the desired output "aaaaa%20aaaa"?

Reply:
AFAIK, 'aaaa' is just arbitrary string. So having `chars[i] = 'a' + (i % 26)` would make the string slightly non-homogenous.
In this case, it won't any difference in benchmark results.